tests/Properties.hs

```diff
@@ -536,5 +545,36 @@ misc_tests =
     , testProperty "readIntegerUnsafe" prop_readIntegerUnsafe
     ]
 
+strictness_checks =
+  [ testGroup "Lazy Word8"
+    [ testProperty "foldr is lazy" $ \ xs ->
+        genericTake (L.length xs) (L.foldr (:) [ ] (explosiveTail xs)) == L.unpack xs
```

Comment:
It does not particulary matter for this case, but I prefer using `(===)` instead of `(==)` in QuickCheck properties, since `(===)` prints left and right sides if they mismatch.

Reply:
Sure. It would be good. I shall fix it.